src/backend/src/services/design-review.services.ts

```diff
@@ -45,6 +52,120 @@ export default class DesignReviewService {
+   */
+  static async createDesignReview(
+    submitter: User,
+    dateScheduled: Date,
+    teamTypeId: string,
+    requiredMemberIds: number[],
+    optionalMemberIds: number[],
+    isOnline: boolean,
+    isInPerson: boolean,
+    docTemplateLink: string,
+    wbsNum: WbsNumber,
+    meetingTimes: number[],
+    zoomLink?: string,
+    location?: string
+  ): Promise<DesignReview> {
+    if (!isLeadership(submitter.role)) throw new AccessDeniedException('create design review');
+
+    const teamType = await prisma.teamType.findFirst({
+      where: { teamTypeId }
+    });
+
+    if (!teamType) {
+      throw new NotFoundException('Team Type', teamTypeId);
+    }
+
+    const wbsElement = await prisma.wBS_Element.findUnique({
+      where: {
+        wbsNumber: {
+          carNumber: wbsNum.carNumber,
+          projectNumber: wbsNum.projectNumber,
+          workPackageNumber: wbsNum.workPackageNumber
+        }
+      }
+    });
+
+    if (!wbsElement) {
+      throw new NotFoundException('WBS Element', wbsNum.carNumber);
+    }
+
+    for (let i = 0; i < meetingTimes.length - 1; i++) {
+      if (i === meetingTimes.length - 1) {
+        continue;
+      }
+      if (meetingTimes[i + 1] - meetingTimes[i] !== 1 || meetingTimes[i] < 0 || meetingTimes[i] > 84) {
+        throw new HttpException(400, 'Meeting times have to be continous and in range 0-48');
+      }
+    }
+
+    if (isOnline && !zoomLink) {
+      throw new HttpException(400, 'If the design review is online then there needs to be a zoom link');
+    }
+
+    if (isInPerson && !location) {
+      throw new HttpException(400, 'If the design review is in person then there needs to be a location');
+    }
+
+    if (dateScheduled.valueOf() <= new Date().valueOf()) {
+      throw new HttpException(400, 'Design review cannot be scheduled for the same day (or before), its created');
+    }
+
+    const designReview = await prisma.design_Review.create({
+      data: {
+        dateScheduled,
+        dateCreated: new Date(),
+        status: Design_Review_Status.UNCONFIRMED,
+        location,
+        isOnline,
+        isInPerson,
+        zoomLink,
+        docTemplateLink,
+        userCreated: { connect: { userId: submitter.userId } },
+        teamType: { connect: { teamTypeId: teamType.teamTypeId } },
+        requiredMembers: { connect: requiredMemberIds.map((memberId) => ({ userId: memberId })) },
+        optionalMembers: { connect: optionalMemberIds.map((memberId) => ({ userId: memberId })) },
+        meetingTimes,
+        wbsElement: { connect: { wbsElementId: wbsElement.wbsElementId } }
+      },
+      ...designReviewQueryArgs
+    });
+
+    // send to all the people invited to the design review
+    for (const memberId of requiredMemberIds.concat(optionalMemberIds)) {
+      const memberUserSettings = await prisma.user_Settings.findUnique({ where: { userId: memberId } });
+      const member = await prisma.user.findUnique({ where: { userId: memberId } });
```

Comment:
you should check that all the memberIds are valid and then fetch an array of the memberUserSettings above, then just loop through the array of memberUserSettings to send the notifications